ci: npm releases

snqb opened the pull request:
Now we can release with [semantic-release](https://github.com/semantic-release/semantic-release)

The proper docs are upcoming

It's working
https://github.com/logicalclocks/quartz/actions
![image](https://user-images.githubusercontent.com/19791699/158791136-3a3ea731-b135-4aa7-98d8-e68a74c48eb3.png)

esonhey commented on .github/workflows/npm-publish.yml:
```diff
@@ -0,0 +1,33 @@
+        with:
+          node-version: 16
+          registry-url: https://registry.npmjs.org/
+      - run: npm ci
+      - run: npm publish
```
If we want to publish it publicly we have to use [`--access public` option](https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#:~:text=If%20your%20npm%20package%20uses%20a%20scope%20prefix%20and%20the%20package%20is%20public%2C%20you%20need%20to%20use%20the%20option%20npm%20publish%20%2D%2Daccess%20public.)

esonhey replied:
Why not use `yarn` instead of `npm` for publishing?

esonhey commented on .github/workflows/npm-publish.yml:
```diff
@@ -0,0 +1,33 @@
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 16
+          registry-url: https://registry.npmjs.org/
```
seems like we are just publishing to npm registry in order to publish it to GitHub Packages we would require to have it as  `'https://npm.pkg.github.com'`

esonhey commented on .github/workflows/npm-publish.yml:
```diff
@@ -0,0 +1,33 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
```
Currently, the configuration publishes to NPM registry instead of GitHub

esonhey commented on .github/workflows/npm-publish.yml:
```diff
@@ -0,0 +1,33 @@
+        with:
+          node-version: 16
+          registry-url: https://registry.npmjs.org/
+      - run: npm ci
```
Since we are installing packages in building project I think we don't need this line

esonhey reviewed (changes requested):
~I wonder why would we need this? 
seems like there is a robot app setup for publishing to GitHub Packages
https://github.com/logicalclocks/quartz/actions/workflows/pages/pages-build-deployment~

Ignore above, it's the bot for GitHub Pages not Packages :sweat_smile:

snqb commented:
@ehsan-github hey, this is a generated thing, never touched anything
all for test purposes :D

esonhey commented:
> @ehsan-github hey, this is a generated thing, never touched anything all for test purposes :D

Ok then let's test ;)

snqb committed:
chore: test release

snqb committed:
chore: update package.json

snqb committed:
chore: test

snqb committed:
chore: a sily attempt at fixing things

snqb committed:
fix: taking it back

snqb committed:
fix: some update config

semantic-release-bot committed:
release(version): Release 1.3.2 [skip ci]

## [1.3.2](https://github.com/logicalclocks/quartz/compare/v1.3.1...v1.3.2) (2022-03-16)

### chore

* a sily attempt at fixing things ([019e8f6](https://github.com/logicalclocks/quartz/commit/019e8f6b7c96d30c07d7c22134d56d1baf442e0b))
* test ([a1eb9d0](https://github.com/logicalclocks/quartz/commit/a1eb9d06609a52975c4248e3cd066d43b8171473))
* test release ([fc45d14](https://github.com/logicalclocks/quartz/commit/fc45d14bcb3b7c71a3e955557537a9cb5823ea6f))
* update package.json ([88405b4](https://github.com/logicalclocks/quartz/commit/88405b48ebe94717a0c36d9e38556072febe50a4))

### fix

* some update config ([cf5aabf](https://github.com/logicalclocks/quartz/commit/cf5aabfab014fe32b7bdaf577704498ea96206fb))
* taking it back ([1d68f7b](https://github.com/logicalclocks/quartz/commit/1d68f7b92c0c72070b57d81d027aa92b5f4eb6c5))

snqb committed:
chore: remove semantic-release-npm-github-publish

snqb committed:
deps: remove semantic-release-npm-github-publish

snqb committed:
chore: change node config

snqb committed:
feat: releases

snqb committed:
chore: leave only `dev` branch

snqb committed:
chore: add whitespace

snqb committed:
refactor: cleanup

snqb committed:
docs: remove CHANGELOG.md

snqb commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,26 @@
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.x'
```
this is our `engines` version

snqb commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,26 @@
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.x'
+          cache: 'yarn'
```
need this to force using `yarn`'s cache

esonhey reviewed (approved):
Really great job!
Looking forward to having [commitizen](https://github.com/commitizen/cz-cli) or [commitlint](https://github.com/conventional-changelog/commitlint) and use them in action
